nalgebra-sparse/src/pattern.rs

```diff
@@ -52,6 +52,17 @@ pub struct SparsityPattern {
     minor_dim: usize,
 }
 
+///The default implementation for the SparsityPattern
+impl Default for SparsityPattern {
+    fn default() -> Self {
+        Self {
+            major_offsets: Default::default(),
+            minor_indices: Default::default(),
+            minor_dim: Default::default(),
+        }
```

Comment:
This is incorrect since `major_offsets` should have length `major_dim + 1`. An "empty" pattern would look like:
```rust
Self {
  major_offsets: vec![0],
  minor_indices: vec![],
  minor_dim: 0
}
```